src/EFCore.PG/Query/Expressions/Internal/PostgresNewArrayExpression.cs

```diff
@@ -0,0 +1,130 @@
+            {
+                var initializer = Initializers[i];
+                var newInitializer = (SqlExpression)visitor.Visit(initializer);
+                if (newInitializer != initializer && newInitializers is null)
+                {
+                    newInitializers = new List<SqlExpression>();
+                    for (var j = 0; j < i; j++)
+                        newInitializers.Add(newInitializer);
+                }
+
+                if (newInitializers != null)
+                    newInitializers.Add(newInitializer);
+            }
+
+            return newInitializers is null
+                ? this
+                : new PostgresNewArrayExpression(newInitializers, Type, TypeMapping);
+        }
+
+        /// <summary>
+        /// Creates a new expression that is like this one, but using the supplied children. If all of the children are the same, it will
+        /// return this expression.
+        /// </summary>
+        /// <param name="initializers">The initializers for the array.</param>
+        /// <returns>This expression if no children changed, or an expression with the updated children.</returns>
+        public virtual PostgresNewArrayExpression Update([NotNull] IReadOnlyList<SqlExpression> initializers)
+        {
+            Check.NotNull(initializers, nameof(initializers));
+
+            return initializers == Initializers
+                ? this
+                : new PostgresNewArrayExpression(initializers, Type, TypeMapping);
+        }
+
+        /// <inheritdoc />
+        public override void Print(ExpressionPrinter expressionPrinter)
+        {
+            Check.NotNull(expressionPrinter, nameof(expressionPrinter));
+
+            expressionPrinter.Append("ARRAY[");
+
+            var first = true;
+            foreach (var initializer in Initializers)
+            {
+                if (!first)
+                {
+                    expressionPrinter.Append(", ");
+                }
+
+                first = false;
+
+                expressionPrinter.Visit(initializer);
+            }
+
+            expressionPrinter.Append("]");
+        }
+
+        /// <inheritdoc />
+        public override bool Equals(object obj)
+            => obj != null
+               && (ReferenceEquals(this, obj)
+                   || obj is PostgresNewArrayExpression sqlBinaryExpression
+                   && Equals(sqlBinaryExpression));
+
+        bool Equals(PostgresNewArrayExpression postgresNewArrayExpression)
+            => base.Equals(postgresNewArrayExpression)
+               && Initializers.SequenceEqual(postgresNewArrayExpression.Initializers);
+
+        /// <inheritdoc />
+        public override int GetHashCode()
+        {
+            var hash = new HashCode();
+            hash.Add(base.GetHashCode());
+            for (var i = 0; i < Initializers.Count; i++)
+                hash.Add(Initializers[i]);
+            return hash.ToHashCode();
```

Comment:
In this method there are no empty lines, but a bit above there are a lot of them. Probably they should be balanced.